src/cpp/src/image_generation/models/unet_inference_static_bs1.hpp

```diff
@@ -0,0 +1,144 @@
+        size_t hidden_states_batch_stride_bytes = encoder_hidden_states.get_strides()[0];
+
+        for (int i = 0; i < m_nativeBatchSize; i++)
+        {
+            auto hidden_states_bs1 = m_requests[i].get_tensor(tensor_name);
+
+            // wrap current pHiddenStates location as batch-1 tensor.
+            ov::Tensor bs1_wrapper(hidden_states_bs1.get_element_type(),
+                                   hidden_states_bs1.get_shape(),
+                                   pHiddenStates,
+                                   encoder_hidden_states.get_strides());
+
+            // copy it to infer request batch-1 tensor
+            bs1_wrapper.copy_to(hidden_states_bs1);
+
+            // increment pHiddenStates to start location of next batch (using stride)
+            pHiddenStates += hidden_states_batch_stride_bytes;
+        }
+    }
+
+    virtual void set_adapters(AdapterController& adapter_controller, const AdapterConfig& adapters) override {
+        OPENVINO_ASSERT(m_nativeBatchSize && m_nativeBatchSize == m_requests.size(),
+                        "UNet model must be compiled first");
+        for (int i = 0; i < m_nativeBatchSize; i++) {
+            adapter_controller.apply(m_requests[i], adapters);
+        }
+    }
+
+    virtual ov::Tensor infer(ov::Tensor sample, ov::Tensor timestep) override {
+        OPENVINO_ASSERT(m_nativeBatchSize && m_nativeBatchSize == m_requests.size(),
+                        "UNet model must be compiled first");
+
+        char* pSample = (char *)sample.data();
+        size_t sample_batch_stride_bytes = sample.get_strides()[0];
+
+        for (int i = 0; i < m_nativeBatchSize; i++) {
+            m_requests[i].set_tensor("timestep", timestep);
+
+            auto sample_bs1 = m_requests[i].get_tensor("sample");
+
+            // wrap current pSample location as batch-1 tensor.
+            ov::Tensor bs1_wrapper(sample_bs1.get_element_type(), sample_bs1.get_shape(), pSample, sample.get_strides());
+
+            // copy it to infer request batch-1 tensor
+            bs1_wrapper.copy_to(sample_bs1);
+
+            //increment pSample to start location of next batch (using stride)
+            pSample += sample_batch_stride_bytes;
+
+            // kick off infer for this request.
+            m_requests[i].start_async();
+        }
+
+        auto out_sample = ov::Tensor(sample.get_element_type(), sample.get_shape());
+
+        char* pOutSample = (char *)out_sample.data();
+        size_t out_sample_batch_stride_bytes = out_sample.get_strides()[0];
+        for (int i = 0; i < m_nativeBatchSize; i++) {
+
+            // wait for infer to complete.
+            m_requests[i].wait();
+
+            auto out_sample_bs1 = m_requests[i].get_tensor("out_sample");
+            ov::Tensor bs1_wrapper(out_sample_bs1.get_element_type(), out_sample_bs1.get_shape(), pOutSample, out_sample.get_strides());
+            out_sample_bs1.copy_to(bs1_wrapper);
```

Comment:
I hope you can avoid copying and ensure that requests writes output data to ROI within "batched" `out_sample`

Reply:
Perhaps here (as opposed to set_hidden_states) we wouldn't cause extra copies by calling 'set_tensor' with bs1 wrappers.. but at the same time we wouldn't be saving any memcpy's either, we'd just be moving where they are taking place. Although this isn't true for other devices... okay, so I'm convinced to refactor this one. :)

Reply:
addressed in this PR: https://github.com/openvinotoolkit/openvino.genai/pull/1172